Endava.TechCourse.BankApp.Application/Commands/CreateWallet/CreateWalletHandler.cs

```diff
@@ -15,19 +16,32 @@ public CreateWalletHandler(ApplicationDbContext context)
 
         public async Task Handle(CreateWalletCommand request, CancellationToken cancellationToken)
         {
+            var user = _context.Users.FirstOrDefault(u => u.Id == new Guid(request.UserId));
             var currency = _context.Currency.FirstOrDefault(c => c.CurrencyCode == request.CurrencyCode);
-            if (currency == null)
+            var type = _context.WalletType.FirstOrDefault(t => t.Id == new Guid(request.TypeId));
+            if (currency == null || user == null || type == null)
             {
-                throw new Exception($"Currency with code '{request.CurrencyCode}' not found.");
+                throw new Exception($"User not found.");
```

Comment:
Check for null right before your did a query to database. And is better to throw a more specific Exception type like InvalidOperationException()